application/src/main/java/org/togetherjava/tjbot/features/moderation/history/PurgeMessageListener.java

```diff
@@ -0,0 +1,116 @@
+package org.togetherjava.tjbot.features.moderation.history;
+
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import org.jooq.impl.UpdatableRecordImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.records.MessageHistoryRecord;
+import org.togetherjava.tjbot.features.MessageReceiverAdapter;
+
+import java.time.Instant;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+import static org.togetherjava.tjbot.db.generated.Tables.MESSAGE_HISTORY;
+
+/**
+ * Listens for new message throughout the guild, then stores some metadata for each message in a
+ * database.
+ */
+public final class PurgeMessageListener extends MessageReceiverAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(PurgeMessageListener.class);
+    private static final int MESSAGES_RECORDS_MAX_LIMIT = 7500;
+    private static final int MESSAGES_RECORDS_THRESHOLD = MESSAGES_RECORDS_MAX_LIMIT - 100;
+    private static final int RECORDS_TO_TRIM = 500;
+    static AtomicInteger recordsCounter = new AtomicInteger(0);
```

Comment:
Zabu suggested to keep DB records in check, not to beyond a limit otherwise it can bring down bot. I chose a limit of 7.5k records which is very very comfortable(can look into server stats for more details). 

`THRESHOLD`-> whenever db records hit 7.4k limit, we clear out last `RECORDS_TO_TRIM` from records table, because if we don't then new messages are not stored and purge kinda becomes useless. This way we always have latest messages to our limit(ideally we should never touch `THRESHOLD` but maybe in future who knows).

Prolly need to work on this, open to ideas/suggestions.